src/Vulkan/RHI/VkPipelineState.cpp

```diff
@@ -8,11 +10,68 @@
 #include <Vulkan/RHI/VkTexture.h>
 #include <Vulkan/VkErrorHandler.h>
 
+#include "../VkMacros.h"
+#include "Vulkan/VkFormats.h"
+
 namespace vex::vk
 {
 
-VkGraphicsPipelineState::VkGraphicsPipelineState(const Key& key)
+namespace GraphicsPiplineUtils
+{
+
+VEX_VK_BEGIN_ENUM_MAPPING(InputTopology, InputTopology, ::vk::PrimitiveTopology, VkTopology)
+VEX_VK_ENUM_MAPPING_ENTRY(TriangleFan, eTriangleFan)
+VEX_VK_ENUM_MAPPING_ENTRY(TriangleList, eTriangleList)
+VEX_VK_ENUM_MAPPING_ENTRY(TriangleStrip, eTriangleStrip)
+VEX_VK_END_ENUM_MAPPING
+
+VEX_VK_BEGIN_ENUM_MAPPING_STATIC(VertexInputLayout::InputRate, InputRate, ::vk::VertexInputRate, VkInputRate)
+VEX_VK_ENUM_MAPPING_ENTRY(PerInstance, eInstance)
+VEX_VK_ENUM_MAPPING_ENTRY(PerVertex, eVertex)
+VEX_VK_END_ENUM_MAPPING
+
+VEX_VK_BEGIN_ENUM_MAPPING_FLAGS_STATIC(CullMode, CullMode, ::vk::CullMode, VkCullMode)
+VEX_VK_ENUM_MAPPING_ENTRY(Back, eBack)
+VEX_VK_ENUM_MAPPING_ENTRY(Front, eFront)
+VEX_VK_ENUM_MAPPING_ENTRY(None, eNone)
+VEX_VK_END_ENUM_MAPPING
+
+VEX_VK_BEGIN_ENUM_MAPPING_STATIC(Winding, Winding, ::vk::FrontFace, VkFrontFace)
+VEX_VK_ENUM_MAPPING_ENTRY(Clockwise, eClockwise)
+VEX_VK_ENUM_MAPPING_ENTRY(CounterClockwise, eCounterClockwise)
+VEX_VK_END_ENUM_MAPPING
+
+VEX_VK_BEGIN_ENUM_MAPPING_STATIC(PolygonMode, PolygonMode, ::vk::PolygonMode, VkPolygonMode)
+VEX_VK_ENUM_MAPPING_ENTRY(Fill, eFill)
+VEX_VK_ENUM_MAPPING_ENTRY(Line, eLine)
+VEX_VK_ENUM_MAPPING_ENTRY(Point, ePoint)
+VEX_VK_END_ENUM_MAPPING
+
+} // namespace GraphicsPiplineUtils
```

Comment:
Could (should?) move this into VkGraphicsPipeline.h/.cpp to match Vex/DX12 file structure.